source/extensions/filters/http/basic_auth/config.cc

```diff
@@ -0,0 +1,73 @@
+#include "source/extensions/filters/http/basic_auth/config.h"
+
+#include "source/common/config/datasource.h"
+#include "source/extensions/filters/http/basic_auth/basic_auth_filter.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace BasicAuth {
+
+using envoy::extensions::filters::http::basic_auth::v3::BasicAuth;
+
+namespace {
+
+UserMapConstPtr readHtpasswd(const std::string& htpasswd) {
+  std::unique_ptr<absl::flat_hash_map<std::string, User>> users =
+      std::make_unique<absl::flat_hash_map<std::string, User>>();
+  std::istringstream htpsswd_ss(htpasswd);
+  std::string line;
+
+  while (std::getline(htpsswd_ss, line)) {
+    const size_t colon_pos = line.find(':');
+
+    if (colon_pos != std::string::npos) {
+      std::string name, hash;
+
+      absl::string_view line_view = line;
```

Comment:
Do we need to use string_view here? I think a plain simple std::string is good enough. @wbpcode

Reply:
Both is OK for this function (because this only be called once when configuration is loaded), although string view still be better to reduce unnecessary sub-string copy/creation.

And please don't declare multiple variables by `std::string name, hash;`.

```
absl::string_view name = line_view.substr(0, colon_pos);
absl::string_view hash = line_view.substr(colon_pos + 1);

if(name.empty()) {
  throw EnvoyException("basic auth: invalid user name");
}

if(!absl::StartsWith(hash_view, "{SHA}")) {
  throw EnvoyException("basic auth: unsupported htpasswd format: please use {SHA}");
}

hash = hash.substr(5);
if (hash.length() != 28) {
  throw EnvoyException("basic auth: invalid SHA hash length");
}

users->emplace(name, User{std::string(name), std::string(hash)});

```

(Not sure whether the compiler will be happy to this code snippet, just as a nit suggestion)